johnzon-mapper/src/main/java/org/apache/johnzon/mapper/internal/DelegatingWriter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *     Licensed to the Apache Software Foundation (ASF) under one or more
+ *     contributor license agreements.  See the NOTICE file distributed with
+ *     this work for additional information regarding copyright ownership.
+ *     The ASF licenses this file to You under the Apache License, Version 2.0
+ *     (the "License"); you may not use this file except in compliance with
+ *     the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+package org.apache.johnzon.mapper.internal;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Objects;
+
+/* A delegating class for a no-close Writer */
+public class DelegatingWriter extends Writer {
+    private Writer writer;
+
+    public DelegatingWriter(Writer writer) {
+        this.writer = Objects.requireNonNull(writer, "writer");
+    }
+
+    @Override
+    public void write(final int c) throws IOException {
+        writer.write(c);
+    }
+
+    @Override
+    public void write(final char[] cbuf) throws IOException {
+        writer.write(cbuf);
+    }
+
+    @Override
+    public void write(final char[] cbuf, final int off, final int len) throws IOException {
+        writer.write(cbuf, off, len);
+    }
+
+    @Override
+    public void write(final String str) throws IOException {
+        writer.write(str);
+    }
+
+    @Override
+    public void write(final String str, final int off, final int len) throws IOException {
+        writer.write(str, off, len);
+    }
+
+    @Override
+    public Writer append(final CharSequence csq) throws IOException {
+        return writer.append(csq);
+    }
+
+    @Override
+    public Writer append(final CharSequence csq, final int start, final int end) throws IOException {
+        return writer.append(csq, start, end);
+    }
+
+    @Override
+    public Writer append(final char c) throws IOException {
+        return writer.append(c);
+    }
+
+    @Override
+    public void flush() throws IOException {
+        writer.flush();
+    }
+
+    // Flushes instead of closing
+    @Override
+    public void close() throws IOException {
```

Comment:
(just explaining for this one but idea is the same for all types) here it should be `writer.*close()*`, then in `Streams` you do`return new DelegatingWriter(writer) { @Override public void close() {flush();} }`;
Idea was just to make most of the impl "standard" and just the specific part "hidden".

side note: even if I'm still not a fan of requireNonNull the minimum would be a valid error message (`writer = requireNonNull(writer, "delegate writer must not be null")`) but as mentionned in comment it is always wrong before in any code path so I would be interested to know how you got it and if the null check wouldn't sit better before in the codepath.

Reply:
I got what you mean. Let me make those changes and notify you then.